src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,151 @@
+        '12' => 'Error in SMS sending',
+        '13' => 'Phone number is in the blackist. SMS sending to this number is forbidden.',
+        '14' => 'There are more than 50 numbers in the request',
+        '15' => 'List not specified',
+        '16' => 'Invalid phone number',
+        '17' => 'SMS ID not specified',
+        '18' => 'Status not obtained',
+        '19' => 'Empty response',
+        '20' => 'The number already exists',
+        '21' => 'No name',
+        '22' => 'Template already exists',
+        '23' => 'Missing Month (Format: YYYY-MM)',
+        '24' => 'Timestamp not specified',
+        '25' => 'Error in access to the list',
+        '26' => 'There are no numbers in the list',
+        '27' => 'No valid numbers',
+        '28' => 'Missing start date (Format: YYYY-MM-DD)',
+        '29' => 'Missing end date (Format: YYYY-MM-DD)',
+        '30' => 'No date (format: YYYY-MM-DD)',
+        '31' => 'Closing direction to the user',
+        '32' => 'Not enough money',
+        '33' => 'Missing phone number',
+        '34' => 'Phone is in stop list',
+        '35' => 'Not enough money',
+        '36' => 'Can not obtain information about phone',
+        '37' => 'Base Id is not set',
+        '38' => 'Phone number already exists in this database',
+        '39' => 'Phone number does not exist in this database',
+    ];
+
+    public function __construct(string $login, string $password, string $phone, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->phone = $phone;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s?phone=%s', $this->getEndpoint(), $this->phone);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $signature = $this->generateSignature($message);
+
+        $endpoint = sprintf(
+            'https://%s/external/get/send.php?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            str_replace('+', '', $message->getPhone()),
+            $message->getSubject(),
+            $this->phone,
+            time()
+        );
+
+        $response = $this->client->request('GET', $endpoint);
+
+        $content = $response->toArray(false);
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        $phone = preg_replace("/[^\d]/", '', $message->getPhone());
```

Comment:
We don't escape phone the phone number the same way here, and when sending the message.

This implementation looks fragile.

Reply:
Fixed.